src/nisarqa/utils/metadata_checks.py

```diff
@@ -168,102 +166,204 @@ def verify_metadata_cubes(
-                        passes &= _check_gdal(c=cube)
-
-                    for cube in product.elevation_antenna_pat_metadata_cubes(
-                        freq
-                    ):
-                        has_finite &= _metadata_cube_has_finite_pixels(cube)
-                        passes &= has_finite
-                        passes &= _metadata_cube_is_not_all_zeros(cube)
-                        passes &= _check_gdal(c=cube)
-
-                    if isinstance(product, nisarqa.RSLC):
-                        for cube in product.geometry_metadata_cubes():
-                            has_finite &= _metadata_cube_has_finite_pixels(cube)
-                            passes &= has_finite
-                            passes &= _metadata_cube_is_not_all_zeros(cube)
-
-                        for cube in product.crosstalk_metadata_cubes():
-                            has_finite &= _metadata_cube_has_finite_pixels(cube)
-                            passes &= has_finite
-                            passes &= _metadata_cube_is_not_all_zeros(cube)
+        for freq in product.freqs:
+
+            if spec < nisarqa.Version(1, 1, 0):
+                break
+
+            for ds in product.metadata_neb_datasets(freq):
+                has_finite &= _dataset_has_finite_pixels(ds)
+                passes &= has_finite
+                passes &= _dataset_has_finite_pixels(ds)
+                passes &= _dataset_is_not_all_zeros(ds)
+                passes &= _check_gdal(product=product, ds=ds)
+
+            for ds in product.metadata_elevation_antenna_pat_datasets(freq):
+                has_finite &= _dataset_has_finite_pixels(ds)
+                passes &= has_finite
+                passes &= _dataset_is_not_all_zeros(ds)
+                passes &= _check_gdal(product=product, ds=ds)
+
+            if isinstance(product, nisarqa.SLC):
+                for ds in product.metadata_geometry_datasets():
+                    has_finite &= _dataset_has_finite_pixels(ds)
+                    passes &= has_finite
+                    passes &= _dataset_is_not_all_zeros(ds)
+                    passes &= _check_gdal(product=product, ds=ds)
+
+            if isinstance(product, nisarqa.RSLC):
+                for ds in product.metadata_crosstalk_datasets():
+                    has_finite &= _dataset_has_finite_pixels(ds)
+                    passes &= has_finite
+                    passes &= _dataset_is_not_all_zeros(ds)
+                summary_notes = ""
+            else:
+                # GSLC and GCOV products contain the `crosstalk` Group with
+                # datasets copied directly from the input RSLC product,
+                # but these are neither geocoded nor georeferenced.
+                # This means that that there is no corresponding
+                # e.g. `xCoordinates` datasets, so it is not possible to
+                # build/test a MetadataDataset.
+                log.warning(
```

Comment:
Fine for now--

In a future PR let's do the following:

* Add an abstract method `metadata_crosstalk_datasets()` to `NonInsarProduct` that returns a `MetadataDatasetT`.
* Add a specialization of this method for `GSLC` and `GCOV` that returns a `MetadataDataset2D` (note this is different from RSLC, which returns a `MetadataDataset1D`). This method will fail for current GSLC/GCOV products, since they don't yet geocode this LUT from the RSLC.
* In the `except` case below, log an error for the bad crosstalk LUTs in GSLC/GCOV products and remove this warning

Reply:
Update: Touched base offline with the team, and it looks like there is still discussion about how to update the GSLC and GCOV `crosstalk` datasets to be useful for the users, such as by georeferencing them. We can still submit an Issue to document this, but there is not a clear path forward yet for the L2 products, and thus also not a clear path forward for QA.